From dc6f0051795e2d74943cbdd5000d4c07a0601374 Mon Sep 17 00:00:00 2001
From: Sasha Levin <sashal@kernel.org>
Date: Wed, 8 Sep 2021 16:41:45 -0500
Subject: ACPICA: Add support for MADT online enabled bit

From: Mario Limonciello <mario.limonciello@amd.com>

[ Upstream commit 435a8dc8d9b9d91e625901fea5b5695b9b976d84 ]

The online enabled bit on newer ACPI implmentations will indicate
whether the CPU is hotpluggable.

Link: http://github.com/acpica/acpica/pull/708/
Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Stable-dep-of: 128b0c9781c9 ("x86/i8259: Skip probing when ACPI/MADT advertises PCAT compatibility")
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 include/acpi/actbl2.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/acpi/actbl2.h b/include/acpi/actbl2.h
index e45ced27f4c30..d385c3ea1396e 100644
--- a/include/acpi/actbl2.h
+++ b/include/acpi/actbl2.h
@@ -716,6 +716,7 @@ struct acpi_madt_generic_translator {
 /* MADT Local APIC flags */
 
 #define ACPI_MADT_ENABLED           (1)	/* 00: Processor is usable if set */
+#define ACPI_MADT_ONLINE_CAPABLE    (2)	/* 01: System HW supports enabling processor at runtime */
 
 /* MADT MPS INTI flags (inti_flags) */
 
-- 
2.42.0

